lightning/src/ln/peers/handshake/mod.rs

```diff
@@ -90,305 +58,186 @@ impl PeerHandshake {
 	/// `.0`: Byte vector containing the next act to send back to the peer per the handshake protocol
-	/// `.1`: Conduit option if the handshake was just processed to completion and messages can now be encrypted and decrypted
-	pub fn process_act(&mut self, input: &[u8]) -> Result<(Option<Act>, Option<Conduit>), String> {
-		let mut response = None;
-		let mut connected_peer = None;
-
-		self.read_buffer.extend_from_slice(input);
-		let read_buffer_length = self.read_buffer.len();
-
-		match &self.state {
-			&Some(HandshakeState::Uninitiated) => {
-				let remote_public_key = &self.remote_public_key.ok_or("outbound connections must be initialized with new_outbound")?;
-				let act_one = self.initiate(&remote_public_key)?;
-				response = Some(Act::One(act_one));
-			}
-			&Some(HandshakeState::AwaitingActOne(_)) => {
-				if read_buffer_length < ACT_ONE_LENGTH {
-					return Err("need at least 50 bytes".to_string());
-				}
-
-				let mut act_one_buffer = [0u8; ACT_ONE_LENGTH];
-				act_one_buffer.copy_from_slice(&self.read_buffer[..ACT_ONE_LENGTH]);
-				self.read_buffer.drain(..ACT_ONE_LENGTH);
-
-				let act_two = self.process_act_one(ActOne(act_one_buffer))?;
-				response = Some(Act::Two(act_two));
-			}
-			&Some(HandshakeState::AwaitingActTwo(_)) => {
-				if read_buffer_length < ACT_TWO_LENGTH {
-					return Err("need at least 50 bytes".to_string());
-				}
-
-				let mut act_two_buffer = [0u8; ACT_TWO_LENGTH];
-				act_two_buffer.copy_from_slice(&self.read_buffer[..ACT_TWO_LENGTH]);
-				self.read_buffer.drain(..ACT_TWO_LENGTH);
-
-				let (act_three, mut conduit) = self.process_act_two(ActTwo(act_two_buffer))?;
-
-				if self.read_buffer.len() > 0 { // have we received more data still?
-					conduit.read(&self.read_buffer[..]);
-					self.read_buffer.drain(..);
-				}
-
-				response = Some(Act::Three(act_three));
-				connected_peer = Some(conduit);
-			}
-			&Some(HandshakeState::AwaitingActThree(_)) => {
-				if read_buffer_length < ACT_THREE_LENGTH {
-					return Err("need at least 66 bytes".to_string());
-				}
-
-				let mut act_three_buffer = [0u8; ACT_THREE_LENGTH];
-				act_three_buffer.copy_from_slice(&self.read_buffer[..ACT_THREE_LENGTH]);
-				self.read_buffer.drain(..ACT_THREE_LENGTH);
-
-				let (public_key, mut conduit) = self.process_act_three(ActThree(act_three_buffer))?;
-
-				if self.read_buffer.len() > 0 { // have we received more data still?
-					conduit.read(&self.read_buffer[..]);
-					self.read_buffer.drain(..);
-				}
-
-				connected_peer = Some(conduit);
-				self.remote_public_key = Some(public_key);
-			}
-			_ => {
-				panic!("no acts left to process");
-			}
+	/// `.1`: Some(Conduit, PublicKey) if the handshake was just processed to completion and messages can now be encrypted and decrypted
+	pub fn process_act(&mut self, input: &[u8]) -> Result<(Option<&[u8]>, Option<(Conduit, PublicKey)>), String> {
+		assert!(self.ready_for_process);
+		let cur_state = self.state.take().unwrap();
```

Comment:
I think we can spell out `current_state` :)

Reply:
Resolved in `review: Non-controversial renames, nits, small fixups`